caster/lib/settings.py

```diff
@@ -35,18 +36,52 @@
 WSR = False
 
 
-def _find_natspeak():
-    '''Tries to find the natspeak engine.'''
-    possible_locations = [
-        "C:/Program Files (x86)/Nuance/NaturallySpeaking15/Program/natspeak.exe",
-        "C:/Program Files (x86)/Nuance/NaturallySpeaking14/Program/natspeak.exe",
-        "C:/Program Files (x86)/Nuance/NaturallySpeaking13/Program/natspeak.exe",
-        "C:/Program Files (x86)/Nuance/NaturallySpeaking12/Program/natspeak.exe",
-    ]
-    for location in possible_locations:
-        if os.path.isfile(location):
-            return location
-    print "Cannot find default dragon engine path"
+def _validate_natspeak():  # Validates 'Engine Path' for DNS in settings.json
+    if os.path.isfile(_SETTINGS_PATH):
+        with io.open(_SETTINGS_PATH, "rt", encoding="utf-8") as json_file:
+            data = json.loads(json_file.read())
+            exe_path = data["paths"]["ENGINE_PATH"]
+            if os.path.isfile(exe_path):
+                return exe_path
+            else:
+                exe_path = _find_natspeak()
+                data["paths"]["ENGINE_PATH"] = exe_path
+                try:
+                    formatted_data = unicode(
+                        json.dumps(data, sort_keys=True, indent=4, ensure_ascii=False))
+                    with io.open(_SETTINGS_PATH, "r+", encoding="utf-8") as json_file:
+                        json_file.write(formatted_data)
+                        print "Setting DNS path to " + exe_path
+                except Exception as e:
+                    print "Error saving json file " + str(e) + _SETTINGS_PATH
+                return ""
```

Comment:
This should return exe_path so that empty string is not set as the default engine path.